qa/integration-tests/src/test/java/io/camunda/it/utils/CamundaMultiDBExtension.java

```diff
@@ -0,0 +1,238 @@
+  }
+
+  @Override
+  public void beforeAll(final ExtensionContext context) {
+    LOGGER.info("Starting up Camunda instance, with {}", databaseType);
+    final Class<?> testClass = context.getRequiredTestClass();
+    testPrefix = testClass.getSimpleName().toLowerCase();
+
+    switch (databaseType) {
+      case LOCAL -> {
+        final ElasticsearchContainer elasticsearchContainer = setupElasticsearch();
+        final String elasticSearchUrl = "http://" + elasticsearchContainer.getHttpHostAddress();
+        validateESConnection(elasticSearchUrl);
+        multiDbConfigurator.configureElasticsearchSupport(elasticSearchUrl, testPrefix);
+      }
+      case ES -> {
+        validateESConnection(DEFAULT_ES_URL);
+        multiDbConfigurator.configureElasticsearchSupport(DEFAULT_ES_URL, testPrefix);
+      }
+      case OS ->
+          multiDbConfigurator.configureOpenSearchSupport(
+              DEFAULT_OS_URL, testPrefix, DEFAULT_OS_ADMIN_USER, DEFAULT_OS_ADMIN_PW);
+      case RDBMS -> multiDbConfigurator.configureRDBMSSupport();
+      default -> throw new RuntimeException("Unknown exporter type");
+    }
+    testApplication.start();
+    testApplication.awaitCompleteTopology();
+
+    injectFields(testClass, null, ModifierSupport::isStatic);
+  }
+
+  private ElasticsearchContainer setupElasticsearch() {
+    final ElasticsearchContainer elasticsearchContainer =
+        TestSearchContainers.createDefeaultElasticsearchContainer();
+    elasticsearchContainer.start();
+    closeables.add(elasticsearchContainer);
+    return elasticsearchContainer;
+  }
+
+  private static void validateESConnection(final String url) {
+    final HttpRequest httpRequest =
+        HttpRequest.newBuilder().GET().uri(URI.create(String.format("%s/", url))).build();
+    try (final HttpClient httpClient = HttpClient.newHttpClient()) {
+      final HttpResponse<String> response = httpClient.send(httpRequest, BodyHandlers.ofString());
+      final int statusCode = response.statusCode();
+      assert statusCode / 100 == 2
+          : "Expected to have a running ES service available under: " + url;
+    } catch (final IOException | InterruptedException e) {
+      assert false
+          : "Expected no exception on validating connection under: "
+              + url
+              + ", failed with: "
+              + e
+              + ": "
+              + e.getMessage();
+    }
+  }
+
+  private void injectFields(
+      final Class<?> testClass, final Object testInstance, Predicate<Field> predicate) {
+    predicate = predicate.and(field -> field.getType() == CamundaClient.class);
+    for (final Field field : testClass.getDeclaredFields()) {
+      try {
+        if (predicate.test(field)) {
+          field.setAccessible(true);
+          field.set(testInstance, createCamundaClient());
+        }
+      } catch (final Exception ex) {
+        throw new RuntimeException(ex);
```

Comment:
:wrench: You can use `LangUtil.rethrowUnchecked` to avoid the unnecessary wrapping.